bin/ruby-build

```diff
@@ -365,14 +365,18 @@ fetch_tarball() {
   if [ "$package_url" != "${package_url/\#}" ]; then
     checksum="${package_url#*#}"
     package_url="${package_url%%#*}"
+  fi
 
-    if [ -n "$RUBY_BUILD_MIRROR_URL" ]; then
-      if [[ -z "$RUBY_BUILD_DEFAULT_MIRROR" || $package_url != */cache.ruby-lang.org/* ]]; then
-        mirror_url="${RUBY_BUILD_MIRROR_URL}/$checksum"
-      fi
-    elif [ -n "$RUBY_BUILD_MIRROR_PACKAGE_URL" ]; then
-      mirror_url="$RUBY_BUILD_MIRROR_PACKAGE_URL"
+  if [ -n "$RUBY_BUILD_IGNORE_CHECKSUM" ]; then
```

Comment:
I do not believe we should expose this additional option, since we would not want anyone to actually use it.

Instead, in order to support the use-case that you are describing (testing Ruby releases that do not yet have their own definition), I suggest that we introduce a new environment variable completely orthogonal to the mirror feature (since this is technically not related to mirroring, either). The new environment variable could be called something like `RUBY_BUILD_TARBALL_OVERRIDE`, and the value read from it would replace both the URL and checksum values in `fetch_tarball`. If the environment variable only supplies a URL, then checksum checks would be off anyway, but a checksum could also be optionally supplied too. How does that sound?

Reply:
Yes that sounds great, I'll update the PR with it.